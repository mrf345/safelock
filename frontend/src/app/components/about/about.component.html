<div class="container d-flex flex-column align-items-center mt-5 pt-2 user-select-none">
  <h1 class="text text-muted pe-none">Safelock version {{ linkService.appVersion$ | async }}</h1>
  <p class="text text-muted text-center mt-3 p-1 pe-none">
    This is a free open-source project licensed <br/>
    under Mozilla Public License version 2.0. For any questions,
    future updates or bug reports please visit
  </p>
  <a (click)="linkService.openUrl('https://github.com/mrf345/safelock')" href="javascript:;">
    github.com/mrf345/safelock/
  </a>
</div>
