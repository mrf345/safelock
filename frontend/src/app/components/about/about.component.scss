@import "../../../styles/variables";

.container {
    animation: $trans-animation;
    animation-duration: $slower-trans-duration;
}
