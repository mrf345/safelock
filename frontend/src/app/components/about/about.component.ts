import { Component } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';

import { LinkService } from '../../services/link.service';

@Component({
  selector: 'app-about',
  standalone: true,
  imports: [CommonModule, RouterModule],
  templateUrl: './about.component.html',
  styleUrl: './about.component.scss'
})
export class AboutComponent {
  constructor(
    readonly linkService: LinkService,
  ) {}
}
