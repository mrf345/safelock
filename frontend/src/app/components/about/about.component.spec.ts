import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
import { of } from 'rxjs';

import { LinkService } from '../../services/link.service';
import { AboutComponent } from './about.component';

describe('AboutComponent', () => {
  let component: AboutComponent;
  let fixture: ComponentFixture<AboutComponent>;

  const version = '12.12.12';

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      imports: [AboutComponent],
      providers: [
        { provide: LinkService, useValue: { appVersion$: of(version) }},
      ],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(AboutComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });


  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match static snapshot', () => {
    expect(fixture.nativeElement).toMatchSnapshot();
  });
});
