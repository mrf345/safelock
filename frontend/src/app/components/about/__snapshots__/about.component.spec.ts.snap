// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutComponent should match static snapshot 1`] = `
<div
  id="root1"
>
  <div
    class="container d-flex flex-column align-items-center mt-5 pt-2 user-select-none"
  >
    <h1
      class="text text-muted pe-none"
    >
      Safelock version 12.12.12
    </h1>
    <p
      class="text text-muted text-center mt-3 p-1 pe-none"
    >
       This is a free open-source project licensed 
      <br />
       under Mozilla Public License version 2.0. For any questions, future updates or bug reports please visit 
    </p>
    <a
      href="javascript:;"
    >
       indexedsolutions.biz/safelock 
    </a>
  </div>
</div>
`;
